src/test/acceptance/base.py

```diff
@@ -58,30 +58,31 @@ def tearDown(self):
         gc.collect()
 
     def _stop_backend(self):
-        with ThreadPoolExecutor(max_workers=4) as pool:
+        with ThreadPoolExecutor(max_workers=5) as pool:
```

Comment:
I'm really not a fan of using a ThreadPoolExecutor to just call multiple methods at once.
Imo the shutdown methods should not be blocking.

But there is nothing to change here, I just wanted to say that I don't like this solution.

Reply:
> Imo the shutdown methods should not be blocking.

The shutdown method needs to block to give the worker processes (which run as independent subprocesses) time to do a clean shutdown. I don't see how we could avoid this (except for simply killing the processes which we don't want).